src/storage-controller/src/instance.rs

```diff
@@ -230,6 +233,22 @@ where
             self.update_paused_statuses();
         }
     }
+
+    /// Refresh the controller state metrics for this instance.
+    ///
+    /// We could also do state metric updates directly in response to state changes, but that would
+    /// mean littering the code with metric update calls. Encapsulating state metric maintenance in
+    /// a single method is less noisy.
+    ///
+    /// This method is invoked by `Controller::maintain`, which we expect to be called once per
+    /// second during normal operation.
```

Comment:
Cound the state be derived from the replica status, too? I think it might be worth leaving a comment to that note, if true. I agree with the implementation that it's easier to maintain the shared bools here than to wire up a SQL query, but we should still explain the design decision if there was any.

Reply:
I don't think so! At least as it's implemented now `mz_cluster_replica_statuses` only has the k8s/pod-level status. That tells us nothing about the application-level status of the replica (e.g. whether the Timely cluster is initialized or whether it's connected to the controller).

Reply:
Maybe we should include the actual connection status somewhere accessible through SQL? Might be worth capturing this in an issue.

Reply:
I added a note to the existing issue about making the status more useful: https://github.com/MaterializeInc/database-issues/issues/4228